### ISSUE TITLE
Using ng12 partial lib and source map causing build memory execption

### ISSUE THREAD

**Issue by rbirkgit** [angular/angular-cli#21779]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Worked with full ivy libs. Stopped working with partial ivy libs

### Description

I have an app with the following source map settings:
```
            "sourceMap": {
              "scripts": true,
              "styles": true,
              "vendor": true
            },
```

I updated to angular 12 and changed my lib to use partial ivy. Now I get memory exception. It works with full ivy.

## 🔬 Minimal Reproduction

See above

## 🔥 Exception or Error

<pre><code>
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF713284C9F napi_wrap+111007
 2: 00007FF713228226 v8::base::CPU::has_sse+59910
 3: 00007FF713229126 node::OnFatalError+294
 4: 00007FF713B02ABE v8::Isolate::ReportExternalAllocationLimitReached+94
 5: 00007FF713AE788D v8::SharedArrayBuffer::Externalize+781
 6: 00007FF7139909DC v8::internal::Heap::EphemeronKeyWriteBarrierFromCode+1516
 7: 00007FF71399BDFA v8::internal::Heap::ProtectUnprotectedMemoryChunks+1258
 </code></pre>

## 🌍 Your Environment

<pre><code>
Angular CLI: 12.2.5
Node: 14.17.6
Package Manager: yarn 1.22.5
OS: win32 x64

Angular: 12.2.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.5
@angular-devkit/build-angular   12.2.5
@angular-devkit/core            12.2.5
@angular-devkit/schematics      12.2.5
@schematics/angular             12.2.5
ng-packagr                      12.2.1
rxjs                            6.6.7
typescript                      4.3.5
</code></pre>

**Anything else relevant?**

No

**Comment by alan-agius4** [comment]
This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by djflex68** [comment]
I'm having the same problem. I do not have a repo (the repo is my whole app indeed) which has about 60 angular libraries modules and many other common/shared modules.
It's hard to setup a small repo here, I think the problem is in fact coming precisely because the repo is big (@rbirkgit might confirm ?)
Setting souceMaps to false makes everything compiling well again.

**Comment by rbirkgit** [comment]
Yes, our library has many modules too (33). I am not sure how how to replicate this. For now we are back using the full ivy mode on our libraries as for us the partial mode is not building. 

We have a demo app in the same repo as the library, and that one compiles, but not the main app in another repo. My guess is that the demo app is not including as many 3rd party vendor scripts? and maybe that's why the main app cannot handle it as it just reaching over the memory limit.

**Comment by alan-agius4** [comment]
Unfortunately, without a reproduction, even shared privately we'll be unable to investigate this issue.

**Comment by djflex68** [comment]
I'm trying to see with my company if it's possible to host the code of our app on a private online repo. This might be complicated because of internal confidentiality policy.

**Comment by djflex68** [comment]
@alan-agius4  I might get this friday a positive answer to share the whole app code on a private repo. I'll keep you in touch. Is that a possible solution ?

**Comment by djflex68** [comment]
I got the approval from my boss to host the app on a private shared repo. I will do that asap.

**Comment by rbirkgit** [comment]
Thanks djflex68! I think it's a good chance we have the same issue.

**Comment by alan-agius4** [comment]
> I will provide this to you tomorrow. Is it possible to send you repo URL per e-mail instead of posting it here ?
> Thanks

Yeah, or since it’s hosted on GitHub you can invite us to the repo via usernames.

**Comment by djflex68** [comment]
I've sent you an e-mail about this ;-)

**Comment by alan-agius4** [comment]
@djflex68, we did get the re-production thanks.

**Comment by rbirkgit** [comment]
Thanks for the update and very exciting!

If it would help and easy to apply, I could also test on my project where we have the same issue.

**Comment by djflex68** [comment]
@JoostK  many thanks for the investigation. I will have a look tomorrow and tell you if that fixed the issue ;-)
